jstz_api/src/encoding/text_decoder.rs

```diff
@@ -0,0 +1,381 @@
+use std::io::Write;
+
+use boa_engine::{
+    js_string,
+    object::{
+        builtins::{JsArrayBuffer, JsDataView, JsInt8Array, JsUint8Array},
+        Object,
+    },
+    property::Attribute,
+    Context, JsArgs, JsError, JsNativeError, JsObject, JsResult, JsString, JsValue,
+    NativeFunction,
+};
+use boa_gc::{Finalize, GcRefMut, Trace};
+use encoding_rs::{CoderResult, Decoder, Encoding};
+use jstz_core::{
+    accessor,
+    native::{
+        register_global_class, Accessor, ClassBuilder, JsNativeObject, NativeClass,
+    },
+    value::{IntoJs, TryFromJs},
+};
+// https://encoding.spec.whatwg.org/#interface-textdecoder
+//
+// dictionary TextDecoderOptions {
+//   boolean fatal = false;
+//   boolean ignoreBOM = false;
+// };
+
+// dictionary TextDecodeOptions {
+//   boolean stream = false;
+// };
+
+// [Exposed=*]
+// interface TextDecoder {
+//   constructor(optional DOMString label = "utf-8", optional TextDecoderOptions Options = {});
+
+//   USVString decode(optional AllowSharedBufferSource input, optional TextDecodeOptions Options = {});
+// };
+// TextDecoder includes TextDecoderCommon;
+
+// https://encoding.spec.whatwg.org/#interface-mixin-textdecodercommon
+// interface mixin TextDecoderCommon {
+//   readonly attribute DOMString encoding;
+//   readonly attribute boolean fatal;
+//   readonly attribute boolean ignoreBOM;
+// };
+
+#[derive(Trace, Finalize)]
+pub struct TextDecoder {
+    encoding: &'static Encoding,
+    #[unsafe_ignore_trace]
+    decoder: Decoder,
+    io_queue: Vec<u8>,
```

Comment:
I know this is more of a spec issue, but it would be good to add some comments here explaining what terms like `io_queue` or `ignore_bom` mean

Reply:
it's explained in the link just above https://encoding.spec.whatwg.org/#interface-mixin-textdecodercommon Is that not enough to just link the spec ?

Reply:
Maybe, at some point we'll enable clippy which ensures comments on everything to generate a internal rustdoc page for us -- so might be good to get into the habit of documenting things